src/python/pants/engine/scheduler.py

```diff
@@ -164,37 +161,36 @@ def _to_ids_buf(self, types):
     def _to_utf8_buf(self, string):
         return self._native.context.utf8_buf(string)
 
-    def _register_rules(self, rule_index):
-        """Record the given RuleIndex on `self._tasks`."""
+    def _register_rules(self, rule_index: RuleIndex) -> Any:
```

Comment:
If this is `Any` because you don't know the more precise type, it's better to leave off the annotation. That way we know to come back to this.

```suggestion
    def _register_rules(self, rule_index: RuleIndex):
```